internal/pkg/manifests/receive/builder.go

```diff
@@ -376,16 +543,67 @@ func ingestorArgsFrom(opts IngesterOptions) []string {
 		fmt.Sprintf("--remote-write.address=0.0.0.0:%d", IngestRemoteWritePort),
 		fmt.Sprintf("--tsdb.path=%s", dataVolumeMountPath),
 		fmt.Sprintf("--tsdb.retention=%s", opts.Retention),
-		`--label=replica="$(NAME)"`,
-		`--label=receive="true"`,
 		fmt.Sprintf("--objstore.config=$(%s)", ingestObjectStoreEnvVarName),
-		fmt.Sprintf("--receive.local-endpoint=$(NAME).%s.$(NAMESPACE).svc.cluster.local:%d",
+		fmt.Sprintf("--receive.local-endpoint=$(POD_NAME).%s.$(POD_NAMESPACE).svc.cluster.local:%d",
 			opts.Name, IngestGRPCPort),
 		"--receive.grpc-compression=none",
 	}
+
+	for k, v := range opts.ExternalLabels {
+		args = append(args, fmt.Sprintf(`--label=%s="%s"`, k, v))
+	}
 	return args
 }
 
+// newHashringConfigMap creates a skeleton ConfigMap for the hashring configuration.
+func newHashringConfigMap(opts HashringOptions) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      opts.Name,
+			Labels:    opts.Labels,
+			Namespace: opts.Namespace,
+		},
+	}
+}
+
+// isExpectedOwner checks if the endpoint slice is owned by the service and if the service is belonged to the hashring.
+func isExpectedOwner(logger logr.Logger, epSlice discoveryv1.EndpointSlice, expectedOwner string) bool {
+	if len(epSlice.GetOwnerReferences()) != 1 {
```

Comment:
Do we need to check for owners at this stage? Given predicate and enqueue did it already for us?

Reply:
So in theory yes!, but this package does not know anything about the existence of predicates and given the importance of the hashrings, I think validating we were passed reasonable info makes sense here, wdyt?

Reply:
Sure works